frontend/src/scenes/error-tracking/queries.test.ts

```diff
@@ -14,75 +14,77 @@ describe('generateSparklineProps', () => {
     })
 
     it('14d', async () => {
-        const labels = getSparklineLabels('14d')
-        expect(labels.length).toEqual(336)
-        expect(labels[0]).toEqual('27 Dec, 2022 18:00 (UTC)')
-        expect(labels[334]).toEqual('10 Jan, 2023 16:00 (UTC)')
-        expect(labels[335]).toEqual('10 Jan, 2023 17:00 (UTC)') // start of hour
-    })
-
-    it('90d', async () => {
-        const labels = getSparklineLabels('90d')
-        expect(labels.length).toEqual(90)
-        expect(labels[0]).toEqual('13 Oct, 2022 00:00 (UTC)')
-        expect(labels[88]).toEqual('9 Jan, 2023 00:00 (UTC)')
-        expect(labels[89]).toEqual('10 Jan, 2023 00:00 (UTC)') // start of day
-    })
-
-    it('180d', async () => {
-        const labels = getSparklineLabels('180d')
-        expect(labels.length).toEqual(26)
-        expect(labels[0]).toEqual('17 Jul, 2022 00:00 (UTC)')
-        expect(labels[24]).toEqual('1 Jan, 2023 00:00 (UTC)')
-        expect(labels[25]).toEqual('8 Jan, 2023 00:00 (UTC)') // start of week
+        const labels = getSparklineLabels('-14d', 4)
+        expect(labels).toEqual([
+            '27 Dec, 2022 17:22 (UTC)',
+            '31 Dec, 2022 05:22 (UTC)',
+            '3 Jan, 2023 17:22 (UTC)',
+            '7 Jan, 2023 05:22 (UTC)',
+        ])
     })
 
     it('mStart', async () => {
-        const labels = getSparklineLabels('mStart')
-        expect(labels.length).toEqual(31)
-        expect(labels[0]).toEqual('11 Dec, 2022 00:00 (UTC)') // goes back one full month
-        expect(labels[29]).toEqual('9 Jan, 2023 00:00 (UTC)')
-        expect(labels[30]).toEqual('10 Jan, 2023 00:00 (UTC)') // start of week
+        const labels = getSparklineLabels('-mStart', 10)
+        expect(labels).toEqual([
+            '1 Jan, 2023 00:00 (UTC)',
+            '1 Jan, 2023 23:20 (UTC)',
+            '2 Jan, 2023 22:40 (UTC)',
+            '3 Jan, 2023 22:00 (UTC)',
+            '4 Jan, 2023 21:20 (UTC)',
+            '5 Jan, 2023 20:41 (UTC)',
+            '6 Jan, 2023 20:01 (UTC)',
+            '7 Jan, 2023 19:21 (UTC)',
+            '8 Jan, 2023 18:41 (UTC)',
+            '9 Jan, 2023 18:01 (UTC)',
+        ])
     })
 
     it('yStart', async () => {
-        const labels = getSparklineLabels('yStart')
-        expect(labels.length).toEqual(52)
-        expect(labels[0]).toEqual('16 Jan, 2022 00:00 (UTC)') // goes back one full year
-        expect(labels[50]).toEqual('1 Jan, 2023 00:00 (UTC)')
-        expect(labels[51]).toEqual('8 Jan, 2023 00:00 (UTC)') // start of week
-    })
-})
-
-describe('constructSparklineConfig', () => {
-    it('arbitrary values', async () => {
-        expect(constructSparklineConfig('4y')).toEqual({ value: 48, interval: 'month' })
-        expect(constructSparklineConfig('10m')).toEqual({ value: 10, interval: 'month' })
-        expect(constructSparklineConfig('6w')).toEqual({ value: 6, interval: 'week' })
+        const labels = getSparklineLabels('yStart', 4)
+        expect(labels).toEqual([
```

Comment:
Epic stuff here! Having the equi-sized stack traces will be a big improvement in terms of making the sparkline feel more consistent. I'm sure some of this maths was a brain teaser